bench/bench_chaum_pedersen.py

```diff
@@ -49,62 +48,64 @@ def identity(x: int) -> int:
     problem_sizes = (100, 500, 1000, 5000)
     rands = Nonces(int_to_q_unchecked(31337))
     speedup: Dict[int, float] = {}
-    print(f"CPUs detected: {cpu_count()}, launching thread pool")
-    pool = Pool(cpu_count())
 
     # warm up the pool to help get consistent measurements
-    results = pool.map(identity, range(1, 30000))
-    assert results == list(range(1, 30000))
-
-    bench_start = timer()
-
-    for size in problem_sizes:
-        print("Benchmarking on problem size: ", size)
-        seeds = rands[0:size]
-        inputs = [
-            BenchInput(
-                get_optional(elgamal_keypair_from_secret(a)),
-                rands[size],
-                rands[size + 1],
-            )
-            for a in seeds
-        ]
-        start_all_scalar = timer()
-        timing_data = [chaum_pedersen_bench(i) for i in inputs]
-        end_all_scalar = timer()
-
-        print(f"  Creating Chaum-Pedersen proofs ({size} iterations)")
-        avg_proof_scalar = average([t[0] for t in timing_data])
-        std_proof_scalar = std([t[0] for t in timing_data])
-        print(f"    Avg    = {avg_proof_scalar:.6f} sec")
-        print(f"    Stddev = {std_proof_scalar:.6f} sec")
-
-        print(f"  Validating Chaum-Pedersen proofs ({size} iterations)")
-        avg_verify_scalar = average([t[1] for t in timing_data])
-        std_verify_scalar = std([t[1] for t in timing_data])
-        print(f"    Avg    = {avg_verify_scalar:.6f} sec")
-        print(f"    Stddev = {std_verify_scalar:.6f} sec")
-
-        # Run in parallel
-        start_all_parallel = timer()
-        timing_data_parallel = pool.map(chaum_pedersen_bench, inputs)
-        end_all_parallel = timer()
-
-        speedup[size] = (end_all_scalar - start_all_scalar) / (
-            end_all_parallel - start_all_parallel
+    with Scheduler() as scheduler:
```

Comment:
this is one way to use the scheduler.  it will automatically be closed at the end of the loop